components/Snackbar/src/MDCSnackbarManager.m

```diff
@@ -244,15 +244,19 @@ - (void)displaySnackbarViewForMessage:(MDCSnackbarMessage *)message {
       showSnackbarView:snackbarView
               animated:YES
             completion:^{
-              if (snackbarView.accessibilityViewIsModal || message.focusOnShow ||
-                  ![self isSnackbarTransient:snackbarView]) {
-                UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification,
-                                                snackbarView);
-              } else {
-                snackbarView.accessibilityElementsHidden = YES;
-                UIAccessibilityPostNotification(UIAccessibilityAnnouncementNotification,
-                                                message.voiceNotificationText);
-              }
+              dispatch_time_t accessibilityPostDelay = dispatch_time(
+                  DISPATCH_TIME_NOW, (int64_t)(message.accessibilityPostDelay * NSEC_PER_SEC));
+              dispatch_after(accessibilityPostDelay, dispatch_get_main_queue(), ^(void) {
```

Comment:
Why do we want the snackbar announcement to be secondary to another announcement? I would expect us to want the snackbar announcement to take precedence. If this is the case, then I'd expect this PR to be structured in the reverse (e.g. providing a way for clients to post their accessibility announcements after snackbar's has finished).

Assuming we do go this route though, delayed accessibility post notifications are a bit of a scary construct (see https://github.com/material-components/material-components-ios/pull/9822) in that it's impossible to confidently know that a delayed announcement will not end up colliding with another announcement. Instead of offering the ability to delay an announcement, can we provide a mechanism for clients to be able to choose when to explicitly invoke the announcement?

Reply:
We want the Snackbar announcement to be the primary announcement and have it persist. Unfortunately, in the client's case, the dismissal of the VC, with it also presents a Snackbar. This causes an initial focus on the Snackbar, and then before it is finished speaking, the initial element of the VC takes focus. I have commented in b/151030466